File: ai-cmo-main/backend/app/models/company.py
import datetime

from sqlmodel import Field, SQLModel

from app.models.types import default_now
from app.settings import settings


class CompanyBase(SQLModel):
    id: int | None = Field(default=None, primary_key=True)
    name: str
    description: str
    name_aliases: str | None = Field(
        default=None, nullable=True
    )  # serialized json array of strings
    website: str
    llm_understanding: str
    products: str | None = Field(default=None, nullable=True)
    is_placeholder: bool = Field(default=False, index=True)
    created_at: datetime.datetime = Field(default_factory=default_now)
    updated_at: datetime.datetime = Field(default_factory=default_now)


if settings.license_type == "ce":

    class Company(CompanyBase, table=True):
        __tablename__ = "companies"  # type: ignore

File: ai-cmo-main/frontend/src/interfaces/company.ts
export interface Company {
    id: number
    account_id: number
    name: string
    description: string
    name_aliases: string | null
    website: string
    llm_understanding: string
    is_placeholder?: boolean
    created_at: Date | string
    updated_at: Date | string
    products?: string
}


export type CompanyConfig = Omit<Company, 'id' | 'account_id' | 'created_at' | 'updated_at' | 'llm_understanding'>